ext/ddtrace_profiling_native_extension/collectors_cpu_and_wall_time_worker.c

```diff
@@ -442,9 +438,9 @@ static VALUE release_gvl_and_run_sampling_trigger_loop(VALUE instance) {
 // This method exists only to enable testing Datadog::Profiling::Collectors::CpuAndWallTimeWorker behavior using RSpec.
 // It SHOULD NOT be used for other purposes.
 static VALUE _native_is_running(DDTRACE_UNUSED VALUE self, VALUE instance) {
-  return \
-    (active_sampler_owner_thread != Qnil && is_thread_alive(active_sampler_owner_thread) && active_sampler_instance == instance) ?
-    Qtrue : Qfalse;
+  struct cpu_and_wall_time_worker_state *state = active_sampler_instance_state; // Read from global variable
+
+  return (state != NULL && is_thread_alive(state->owner_thread) && state->self_instance == instance) ? Qtrue : Qfalse;
```

Comment:
I was wondering about the thread safety of this check. Line 255 has a comment and it seems to be the only area where mutation happens.

I would suggest annotating this with a comment though.

Reply:
I've clarified it in c7f25c3f1e2ecf8a3bcd5a90f0557e7582eee683; let me know if that's still not clear.